### PR TITLE
ci: Add Github Pages Action

### PR DESCRIPTION
**Pull request by krille-chan**
This should make it more easy to push to github pages in all repostiories

### CODE REVIEW

File: .github/workflows/github-pages.yml [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+        type: string
+        required: true
+
+jobs:
+  publish:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
```
Comment (techno-disaster) [inline .github/workflows/github-pages.yml:16]: v3 maybe?

File: .github/workflows/github-pages.yml [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+      - name: Setup Git
+        run: |
+          git config --local user.email "action@github.com"
+          git config --local user.name "GitHub Action"
+
+      - name: Publish to GitHub Pages
+        run: |
+          git checkout --orphan gh-pages
+          rm -rf ./*
+          cp -r ./${{ github.event.inputs.subdirectory }}/* ./
```
Comment (techno-disaster) [inline .github/workflows/github-pages.yml:27]: just `inpits.directory` should work right?

File: .github/workflows/github-pages.yml (hunk 1 above)
Comment (krille-chan) [reply .github/workflows/github-pages.yml]: Fixed

File: .github/workflows/github-pages.yml (hunk 2 above)
Comment (krille-chan) [reply .github/workflows/github-pages.yml]: I have changed it to using download artifacts

**Comment by techno-disaster** [comment]
A flag to delete the artifact later would be nice

File: .github/workflows/github-pages.yml [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
+jobs:
+  publish:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+      - name: Clean repository
+        run: rm -rf ./*
```
Comment (techno-disaster) [inline .github/workflows/github-pages.yml:18]: Do we need this?

Comment (krille-chan) [reply .github/workflows/github-pages.yml:18]: I think yes. This makes sure that we do not push the whole repository to github pages as github pages takes everything from a specific branch and not a subdirectory

Comment (techno-disaster) [reply .github/workflows/github-pages.yml:18]: ah ok I was confused why we need to download the repo and then delete it anyway, looks like its needed by download artifact.

(also nitpick actions/download-artifact@ v3 exists) :p

Comment (krille-chan) [reply .github/workflows/github-pages.yml:18]: Changed :)

**Review by techno-disaster — APPROVED** [review]

**Pull request merged**